### PR TITLE
Streamline 643: Postgresql support

### PR DESCRIPTION
**Pull request by arunmahadevan**
Fixes on top of Harsha's patch https://github.com/hortonworks/streamline/pull/502

Tested bootstrap, topology creation and deployment with different components and works fine.

@harshach please review, also you may close your PR since this patch builds on top of those changes.

### CODE REVIEW

File: bootstrap/sql/postgresql/create_tables.sql [hunk 1]
```diff
@@ -0,0 +1,344 @@
+-- THE NAMES OF THE TABLE COLUMNS MUST MATCH THE NAMES OF THE CORRESPONDING CLASS MODEL FIELDS;
+
+CREATE TABLE IF NOT EXISTS dashboard (
```
Comment (priyank5485) [inline bootstrap/sql/postgresql/create_tables.sql]: In one of the other PRs you have renamed these tables and few classes to a standard. Do we want to do that here too?

File: storage/core/src/main/java/com/hortonworks/streamline/storage/impl/jdbc/JdbcStorageManager.java [hunk 2]
```diff
@@ -233,6 +234,9 @@ public void init(Map<String, Object> properties) {
             case "mysql":
```
Comment (priyank5485) [inline storage/core/src/main/java/com/hortonworks/streamline/storage/impl/jdbc/JdbcStorageManager.java]: Do we still want to keep mysql here?

File: storage/tool/src/main/java/com/hortonworks/streamline/storage/tool/StorageProviderConfiguration.java [hunk 3]
```diff
@@ -38,6 +40,7 @@ private StorageProviderConfiguration(String driverClass, String url, String user
         this.delimiter = delimiter;
     }
 
+
     public static StorageProviderConfiguration mysql(String driverClass, String url) {
```
Comment (priyank5485) [inline storage/tool/src/main/java/com/hortonworks/streamline/storage/tool/StorageProviderConfiguration.java]: Should we remove mysql related varialbes and methods?

File: storage/tool/src/main/java/com/hortonworks/streamline/storage/tool/StorageProviderConfigurationReader.java [hunk 4]
```diff
@@ -103,4 +108,24 @@ private static StorageProviderConfiguration readPhoenixProperties(Map<String, Ob
...
+     * storageProviderConfiguration:
+     *   providerClass: "com.hortonworks.streamline.storage.impl.jdbc.JdbcStorageManager"
+     *   properties:
+     *     db.type: "postgresql"
+     *     queryTimeoutInSecs: 30
+     *     db.properties:
+     *     dataSourceClassName: "org.postgresql.ds.PGSimpleDataSource"
+     *     dataSource.url: "jdbc:postgresql://localhost/test"
+     *     dataSource.user: "postgres"
+     *     dataSource.password: "postgres"
```
Comment (priyank5485) [inline storage/tool/src/main/java/com/hortonworks/streamline/storage/tool/StorageProviderConfigurationReader.java]: Should we add this commented configuration to yaml file?

**Comment by priyank5485** [comment]
+1 once the comments are addressed.

File: bootstrap/sql/postgresql/create_tables.sql (hunk 1 above)
Comment (arunmahadevan) [inline bootstrap/sql/postgresql/create_tables.sql]: Old names were used for compatibility with Pojos and for testing. After https://github.com/hortonworks/streamline/pull/512 is merged, I will rename these.

File: storage/tool/src/main/java/com/hortonworks/streamline/storage/tool/StorageProviderConfiguration.java (hunk 3 above)
Comment (arunmahadevan) [inline storage/tool/src/main/java/com/hortonworks/streamline/storage/tool/StorageProviderConfiguration.java]: I think we still want to keep mysql support, so we can retain this.

File: storage/tool/src/main/java/com/hortonworks/streamline/storage/tool/StorageProviderConfigurationReader.java (hunk 4 above)
Comment (arunmahadevan) [inline storage/tool/src/main/java/com/hortonworks/streamline/storage/tool/StorageProviderConfigurationReader.java]: Similar config was earlier there but was removed from the yaml file so I did not add it again.

File: storage/core/src/main/java/com/hortonworks/streamline/storage/impl/jdbc/JdbcStorageManager.java (hunk 2 above)
Comment (arunmahadevan) [inline storage/core/src/main/java/com/hortonworks/streamline/storage/impl/jdbc/JdbcStorageManager.java]: I think we still want to keep mysql support, so we can retain this.

**Comment by arunmahadevan** [comment]
@priyank5485 after #512 is merged, I will rename the postgres tables and update this PR. Then we can merge this.

**Comment by HeartSaVioR** [comment]
Would we want to add `streamline.postgre.yaml` so that users can start from copying and modifying?

File: bootstrap/sql/postgresql/create_tables.sql (hunk 1 above)
Comment (priyank5485) [inline bootstrap/sql/postgresql/create_tables.sql]: PR 512 is merged. Did you rename these in this PR?

File: storage/core/src/main/java/com/hortonworks/streamline/storage/impl/jdbc/JdbcStorageManager.java (hunk 2 above)
Comment (priyank5485) [inline storage/core/src/main/java/com/hortonworks/streamline/storage/impl/jdbc/JdbcStorageManager.java]: Ok. In that case there was some other place in code where mysql was removed as part of changeset of this PR.  I think it was an "if" check which enforced that only postgre is supported. Not sure if we need to change that but can you can double check we have it right to support mysql?

**Comment by priyank5485** [comment]
@HeartSaVioR Yes. We should add streamline.postgre.yaml file.

**Comment by arunmahadevan** [comment]
Closing this PR since new commits are not getting updated here.

Raised, https://github.com/hortonworks/streamline/pull/514

**Pull request closed without merging**